overlay/agent.cpp

```diff
@@ -351,7 +351,7 @@ void ManagerProcess::updateAgentOverlays(
   // duplicate update corresponding to one already in progress. We
   // should therefore not setup a response for acknowledging this
   // registration.
-  if (futures.empty()) {
+  if (message.overlays_size() > 0 && futures.empty()) {
```

Comment:
Didn't understand why we need this additional condition?

Reply:
We expect to get an empty futures list here if there is no overlay network configured or all of them are disabled.